### PR TITLE
Demo Segmenter Fix

### PR DESCRIPTION
**Pull request by ambiguousname**
Builds on https://github.com/unicode-org/icu4x/pull/5696

### CODE REVIEW

File: ffi/capi/bindings/demo_gen_custom/WordSegmenter.mjs
Comment (robertbastian) [inline ffi/capi/bindings/demo_gen_custom/WordSegmenter.mjs:1]: So this is marked as generated code but it's not. Please remove non-generated code from `ffi/capi/bindings`

**Review by robertbastian — APPROVED** [review]

File: ffi/capi/bindings/demo_gen_custom/WordSegmenter.mjs
Comment (robertbastian) [reply ffi/capi/bindings/demo_gen_custom/WordSegmenter.mjs:1]: Please move to a different folder instead

Comment (ambiguousname) [reply ffi/capi/bindings/demo_gen_custom/WordSegmenter.mjs:1]: Let me know if the new folder looks okay, moved it into the `src` folder

Comment (robertbastian) [reply ffi/capi/bindings/demo_gen_custom/WordSegmenter.mjs:1]: I'd prefer it if lived in the npm folder, as it's JS code and only relevant to the demo.

Comment (ambiguousname) [reply ffi/capi/bindings/demo_gen_custom/WordSegmenter.mjs:1]: Alright, let me know if that looks good

**Review by robertbastian — APPROVED** [review]

**Pull request merged**